### PR TITLE
Fix radiasoft/raydata#3: User uploads masks

### PR DESCRIPTION
**Pull request by e-carlin**
This is based on sirepo-3867-analysis-png please review and merge #3889 before reviewing this.

### CODE REVIEW

File: sirepo/template/raydata.py [hunk 1]
```diff
@@ -66,6 +104,14 @@ def write_parameters(data, run_dir, is_parallel):
...
         _generate_parameters_file(data),
     )
+    m = data.models.inputFiles.mask
+    if m:
+        # SECURITY: extractall protects against malicious filenames
+        zipfile.ZipFile(run_dir.join(_SIM_DATA.lib_file_name_with_model_field(
+            'inputFiles',
+            'mask',
+            m,
+        ))).extractall(path=_MASK_PATH)
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/template/raydata.py]: d788ac657

**Comment by e-carlin** [comment]
@robnagler ready for another review

File: sirepo/util.py [hunk 2]
```diff
@@ -280,6 +282,29 @@ def random_base62(length=32):
...
+    Args:
+      path_or_bytes (py.path or str or bytes): The path to the archive or it's contents
+
+    Returns:
+       (py.path, bytes): The basename of the file, the contents of the file
+    """
+    p = path_or_bytes
+    if isinstance(p, bytes):
+        p = six.BytesIO(p)
+    with zipfile.ZipFile(p, 'r') as z:
```
Comment (robnagler) [inline sirepo/util.py:299]: I would comment that ZipFile does some protection.

File: sirepo/util.py [hunk 3]
```diff
@@ -280,6 +282,29 @@ def random_base62(length=32):
...
+        for i in z.infolist():
+            if i.is_dir():
+                continue
+            # SECURITY: Use only basename of file to prevent against
```
Comment (robnagler) [inline sirepo/util.py:303]: great.

**Review by robnagler — APPROVED** [review]

**Pull request merged**